### ISSUE TITLE
Turn on `noImplicitAny` and `strictPropertyInitialization` for `fast-router`

### ISSUE THREAD

**Issue by chrisdholt** [Microsoft/fast-dna#5341]
As part of #5303 we turned off @typescript-eslint/typedef globally as it was overwritten across multiple packages. As part of that, we turned on `noImplicitAny` and `strictPropertyInitialization` for all packages. Turning this on revealed several issues related to both tsconfig settings in `fast-router` and so it was turned off explicitly for that package. The errors should be fixed and addressed as part of this issue and the code should be tested to ensure that there are no breaks or side effects of the change. This issue exists to track turning that back on and resolving the errors.

**Comment by ritikBhandari** [comment]
Hi @chrisdholt,
So as far as I can understand, while turning on `noImplicitAny` and `strictPropertyInitialization`, the issues related to both tsconfig settings in `fast-router` also need to be resolved. Right?

**Comment by chrisdholt** [comment]
> Hi @chrisdholt, So as far as I can understand, while turning on `noImplicitAny` and `strictPropertyInitialization`, the issues related to both tsconfig settings in `fast-router` also need to be resolved. Right?

Yes, the goal for this would be to turn those settings on and resolve the issues related to them for the `fast-router` package.

**Comment by ritikBhandari** [comment]
Would u mine if I try to resolve this issue?

**Comment by ritikBhandari** [comment]
> Turning this on revealed several issues related to both tsconfig settings in fast-router

@chrisdholt how should I reproduce these errors again? Did u find it in testing phase or any other case?

**Comment by chrisdholt** [comment]
> > Turning this on revealed several issues related to both tsconfig settings in fast-router
> 
> @chrisdholt how should I reproduce these errors again? Did u find it in testing phase or any other case?

If I recall, the issue should repro by turning on the settings in tsconfig and running the build and tests steps.

**Comment by ritikBhandari** [comment]
During testing, all test cases are running except for `fast-router`. Is that the issue?

**Comment by chrisdholt** [comment]
> During testing, all test cases are running except for `fast-router`. Is that the issue?

No the issue is you would see errors thrown relating to instances of no-implicit-any and several cases where strictPropertyInitialization isn’t being done as expected. Not at my computer but I can try to give an example tomorrow (maybe this evening but will be afk for a bit).

**Comment by ritikBhandari** [comment]
Sure.
In the meanwhile i'll check why the tests are running with exception.

**Comment by ritikBhandari** [comment]
@chrisdholt can you attach any example of the errors as I still cannot find them after turning on those parameters.

**Comment by ritikBhandari** [comment]
![fasterror](https://user-images.githubusercontent.com/69508348/160252657-a6def512-b116-417a-a816-62cd43605c6c.png)

Are we noticing these type of errors?

**Comment by chrisdholt** [comment]
> ![fasterror](https://user-images.githubusercontent.com/69508348/160252657-a6def512-b116-417a-a816-62cd43605c6c.png)
> 
> @chrisdholt Are we noticing these type of errors?

That looks like what I would reproduce - those appear to be `noImplicitAny` errors.

**Comment by ritikBhandari** [comment]
So now we're sure these are the errors to be fixed right?

**Comment by chrisdholt** [comment]
> So now we're sure these are the errors to be fixed right?

Yes, we want to fix the `noImplicitAny` typings, which is likely going to require adding the correct typings. That could open up potential type issues with the implementation where something may be inferred but could be undefined or may not overlap. 

Probably best to get @EisenbergEffect to confirm, but I think a PR here for stricter typing would be an improvement, it'll likely just take some time to review depending on what all gets flagged as you move through fixing these.

**Comment by EisenbergEffect** [comment]
The router is still in a bit of a volatile stage so I haven't added all the typings. It can be easier to evolve something at first without that, adding it later once the code has matured. That said, I'd definitely take a PR to improve this now.

**Comment by NiteshSethi09** [comment]
Hey, is there anyone working on this issue? I'd like to give a try. @EisenbergEffect @chrisdholt
